### PR TITLE
C-Blosc: bump

### PR DESCRIPTION
**Pull request by extrowerk**

### CODE REVIEW

File: dev-libs/c_blosc/c_blosc-1.13.7.recipe [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+CHECKSUM_SHA256="a27005c242a496f7a208fe0a2c6bf2a6956b0d30e87fb1217a5f33e807e90dde"
+SOURCE_DIR="c-blosc-$portVersion"
+PATCHES="c_blosc-$portVersion.patchset"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	c_blosc$secondaryArchSuffix = $portVersion compat >= 1
+	lib:libblosc$secondaryArchSuffix = 1.13.7 compat >= 1
```
Comment (alaviss) [inline dev-libs/c_blosc/c_blosc-1.13.7.recipe]: How about using `libVersionCompat`? Also, I think we could just drop `compat` for `c_blosc`, as we only have version >= 1 in HaikuPorts.

```diff
 ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
 SECONDARY_ARCHITECTURES="x86"
 
+libVersion="$portVersion"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
 PROVIDES="
-	c_blosc$secondaryArchSuffix = $portVersion compat >= 1
-	lib:libblosc$secondaryArchSuffix = 1.13.7 compat >= 1
+	c_blosc$secondaryArchSuffix = $portVersion
+	lib:libblosc$secondaryArchSuffix = $libVersionCompat
 "
```

File: dev-libs/c_blosc/c_blosc-1.13.7.recipe [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	haiku$secondaryArchSuffix
+	lib:liblz4$secondaryArchSuffix
+	lib:libsnappy$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libzstd$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	c_blosc${secondaryArchSuffix}_devel = $portVersion
+	devel:libblosc${secondaryArchSuffix} = 1.13.7 compat >= 1
```
Comment (alaviss) [inline dev-libs/c_blosc/c_blosc-1.13.7.recipe]:
```diff
-	devel:libblosc${secondaryArchSuffix} = 1.13.7 compat >= 1
+	devel:libblosc$secondaryArchSuffix = $libVersionCompat

File: dev-libs/c_blosc/c_blosc-1.13.7.recipe [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	cd haiku_build
+
+	cmake .. \
+		-DPREFER_EXTERNAL_LZ4=ON \
+		-DPREFER_EXTERNAL_SNAPPY=ON \
+		-DPREFER_EXTERNAL_ZLIB=ON \
+		-DPREFER_EXTERNAL_ZSTD=ON \
+		$cmakeDirArgs
+
+	make ${jobArgs}
```
Comment (alaviss) [inline dev-libs/c_blosc/c_blosc-1.13.7.recipe]:
```diff
-	make ${jobArgs}
+	make $jobArgs

File: dev-libs/c_blosc/c_blosc-1.13.7.recipe [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	lib:libz$secondaryArchSuffix
+	lib:libzstd$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	c_blosc${secondaryArchSuffix}_devel = $portVersion
+	devel:libblosc${secondaryArchSuffix} = 1.13.7 compat >= 1
+	"
+REQUIRES_devel="
+	c_blosc${secondaryArchSuffix} == $portVersion base
```
Comment (alaviss) [inline dev-libs/c_blosc/c_blosc-1.13.7.recipe]:
```diff
-	c_blosc${secondaryArchSuffix} == $portVersion base
+	c_blosc$secondaryArchSuffix == $portVersion base

File: dev-libs/c_blosc/c_blosc-1.13.7.recipe [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+	"
+
+BUILD_REQUIRES="
+	devel:liblz4$secondaryArchSuffix
+	devel:libsnappy$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:libzstd$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	haiku${secondaryArchSuffix}_devel
```
Comment (alaviss) [inline dev-libs/c_blosc/c_blosc-1.13.7.recipe]:
```diff
 BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
 	devel:liblz4$secondaryArchSuffix
 	devel:libsnappy$secondaryArchSuffix
 	devel:libz$secondaryArchSuffix
 	devel:libzstd$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
-	haiku${secondaryArchSuffix}_devel
-	cmd:gcc${secondaryArchSuffix}
-	cmd:ld${secondaryArchSuffix}
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
-	cmd:cmake

File: dev-libs/c_blosc/c_blosc-1.13.7.recipe [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+memcpy() OS call.  Blosc is the first compressor (that I'm aware of) that is \
+meant not only to reduce the size of large datasets on-disk or in-memory, but \
+also to accelerate memory-bound computations."
+HOMEPAGE="http://www.blosc.org"
+COPYRIGHT="2012-2017 Francesc Alted"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/Blosc/c-blosc/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="a27005c242a496f7a208fe0a2c6bf2a6956b0d30e87fb1217a5f33e807e90dde"
+SOURCE_DIR="c-blosc-$portVersion"
```
Comment (alaviss) [inline dev-libs/c_blosc/c_blosc-1.13.7.recipe]: Per [Guidelines](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#archive-name)

```diff
 CHECKSUM_SHA256="a27005c242a496f7a208fe0a2c6bf2a6956b0d30e87fb1217a5f33e807e90dde"
+SOURCE_FILENAME="c_blosc-$portVersion.tar.gz"
 SOURCE_DIR="c-blosc-$portVersion"

File: dev-libs/c_blosc/c_blosc-1.13.7.recipe [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+	haiku$secondaryArchSuffix
+	lib:liblz4$secondaryArchSuffix
+	lib:libsnappy$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libzstd$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	c_blosc${secondaryArchSuffix}_devel = $portVersion
+	devel:libblosc${secondaryArchSuffix} = $libVersionCompat
```
Comment (fbrosson) [inline dev-libs/c_blosc/c_blosc-1.13.7.recipe]:
```diff
-	devel:libblosc${secondaryArchSuffix} = $libVersionCompat
+	devel:libblosc$secondaryArchSuffix = $libVersionCompat
```

File: dev-libs/c_blosc/c_blosc-1.13.7.recipe [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+	haiku${secondaryArchSuffix}_devel
+	devel:liblz4$secondaryArchSuffix
+	devel:libsnappy$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:libzstd$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:gcc${secondaryArchSuffix}
+	cmd:ld${secondaryArchSuffix}
```
Comment (fbrosson) [inline dev-libs/c_blosc/c_blosc-1.13.7.recipe]:
```diff
-	cmd:gcc${secondaryArchSuffix}
+	cmd:gcc${secondaryArchSuffix}
-	cmd:ld${secondaryArchSuffix}
+	cmd:ld$secondaryArchSuffix
```

File: dev-libs/c_blosc/c_blosc-1.13.7.recipe [hunk 9]
```diff
@@ -0,0 +1,99 @@
+SUMMARY="Blocking, shuffling and lossless compression library"
+DESCRIPTION="Blosc is a high performance compressor optimized for binary \
+data. It has been designed to transmit data to the processor cache faster \
+than the traditional, non-compressed, direct memory fetch approach via a \
+memcpy() OS call. Blosc is the first compressor (that I'm aware of) that is \
+meant not only to reduce the size of large datasets on-disk or in-memory, but \
+also to accelerate memory-bound computations."
+HOMEPAGE="http://www.blosc.org"
```
Comment (fbrosson) [inline dev-libs/c_blosc/c_blosc-1.13.7.recipe]:
```diff
-HOMEPAGE="http://www.blosc.org"
+HOMEPAGE="http://www.blosc.org/"
```

File: dev-libs/c_blosc/patches/c_blosc-1.13.7.patchset [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+diff --git a/tests/test_common.h b/tests/test_common.h
+index 4e40533..7ed56f7 100644
+--- a/tests/test_common.h
++++ b/tests/test_common.h
+@@ -68,7 +68,7 @@ static void* blosc_test_malloc(const size_t alignment, const size_t size)
+ #elif defined(_WIN32)
+   /* A (void *) cast needed for avoiding a warning with MINGW :-/ */
+   block = (void *)_aligned_malloc(size, alignment);
+-#elif _POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600
++#elif _POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600 || defined(__HAIKU__)
```
Comment (alaviss) [inline dev-libs/c_blosc/patches/c_blosc-1.13.7.patchset]: `_POSIX_C_SOURCE` should be defined by the application itself. You should look around to see if any header used by the program does so. If not, you should patch it as to conform the [POSIX specification][0].

> #### The _POSIX_C_SOURCE Feature Test Macro
> 
> A POSIX-conforming application should ensure that the feature test macro _POSIX_C_SOURCE is defined before inclusion of any header.
> 
> When an application includes a header described by IEEE Std 1003.1-2001, and when this feature test macro is defined to have the value 200112L:
> 
> 1. All symbols required by IEEE Std 1003.1-2001 to appear when the header is included shall be made visible.
> 
> 2. Symbols that are explicitly permitted, but not required, by IEEE Std 1003.1-2001 to appear in that header (including those in reserved name spaces) may be made visible.
> 
> 3. Additional symbols not required or explicitly permitted by IEEE Std 1003.1-2001 to be in that header shall not be made visible, except when enabled by another feature test macro.
> 
> Identifiers in IEEE Std 1003.1-2001 may only be undefined using the **#undef** directive as described in Use and Implementation of Functions or The Name Space. These **#undef** directives shall follow all **#include** directives of any header in IEEE Std 1003.1-2001.

[0]: http://pubs.opengroup.org/onlinepubs/007904975/functions/xsh_chap02_02.html

Comment (fbrosson) [reply dev-libs/c_blosc/patches/c_blosc-1.13.7.patchset]: @alaviss, would you say that a correct upstream patch would be to insert this bloc:
```
#ifdef __HAIKU__
#define _POSIX_C_SOURCE 200112L
#endif /* defined(__HAIKU__) */
```
in a header that is included before the inclusion of `test_common.h` ?
BTW, would you say that each compilation unit should see this bloc, even if does not include `test_common.h`?

Comment (alaviss) [reply dev-libs/c_blosc/patches/c_blosc-1.13.7.patchset]: Found this CMake [snippet][0] that would be interesting as a patch. And yes, this is an upstream issue that should be fixed.

[0]: https://github.com/kcat/openal-soft/commit/4e6acfa0d549281efcbe8293d95b76a37deb370d#diff-af3b638bc2a3e6c650974192a53c7291R102

**Pull request merged**